File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/hunters_aura/HuntersAuraDataSerializer.java
package aurocosh.divinefavor.common.custom_data.player.data.hunters_aura;

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer;
import net.minecraft.nbt.NBTTagCompound;

// Handles the actual read/write of the nbt.
public class HuntersAuraDataSerializer implements INbtSerializer<HuntersAuraData> {
    private static final String TAG_HUNTERS_CHANCE = "HuntersChance";

    @Override
    public void serialize(NBTTagCompound nbt, HuntersAuraData instance) {
        nbt.setFloat(TAG_HUNTERS_CHANCE, instance.getChance());
    }

    @Override
    public void deserialize(NBTTagCompound nbt, HuntersAuraData instance) {
        instance.setChance(nbt.getFloat(TAG_HUNTERS_CHANCE));
    }
}

File: src/main/java/aurocosh/divinefavor/common/custom_data/player/data/aura/hunters/HuntersAuraDataSerializer.kt
package aurocosh.divinefavor.common.custom_data.player.data.aura.hunters

import aurocosh.divinefavor.common.lib.interfaces.INbtSerializer
import net.minecraft.nbt.NBTTagCompound

// Handles the actual read/write of the nbt.
class HuntersAuraDataSerializer : INbtSerializer<HuntersAuraData> {

    override fun serialize(nbt: NBTTagCompound, instance: HuntersAuraData) {
        nbt.setFloat(TAG_HUNTERS_CHANCE, instance.chance)
    }

    override fun deserialize(nbt: NBTTagCompound, instance: HuntersAuraData) {
        instance.chance = nbt.getFloat(TAG_HUNTERS_CHANCE)
    }

    companion object {
        private const val TAG_HUNTERS_CHANCE = "HuntersChance"
    }
}
